test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildRestoreUtilityTests.cs

```diff
@@ -2655,6 +2655,377 @@ public void MSBuildRestoreUtility_AddPackageDownloads_InvalidVersionRange(string
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                });
+
+                // Central Version for the package above and another one for a package y
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "CentralPackageVersion" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "Id", "x" },
+                    { "VersionRange", "1.0.0-beta.*" },
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                });
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "CentralPackageVersion" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "Id", "y" },
+                    { "VersionRange", "2.0.0-beta.*" },
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                });
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "CentralPackageVersion" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "Id", "z" },
+                    { "VersionRange", "3.0.0" },
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                });
+
+                var wrappedItems = items.Select(CreateItems).ToList();
+
+                // Act
+                var dgSpec = MSBuildRestoreUtility.GetDependencySpec(wrappedItems);
+                var project1Spec = dgSpec.Projects.Single(e => e.Name == projectName);
+               
+                // Assert
+                Assert.Equal(1, project1Spec.TargetFrameworks.Count());
+                Assert.Equal(2, project1Spec.TargetFrameworks.First().Dependencies.Count);
+                Assert.Equal(3, project1Spec.TargetFrameworks.First().CentralPackageVersions.Count);
+
+                var dependencyX = project1Spec.TargetFrameworks.First().Dependencies.Where(d => d.Name == "x").First();
+                var dependencyY = project1Spec.TargetFrameworks.First().Dependencies.Where(d => d.Name == "y").First();
+
+                Assert.Equal("[1.0.0-beta.*, )", dependencyX.LibraryRange.VersionRange.ToNormalizedString());
+                Assert.Equal(LibraryIncludeFlags.Compile | LibraryIncludeFlags.Build, dependencyX.IncludeType);
+                Assert.Equal("[1.2.1, )", dependencyY.LibraryRange.VersionRange.ToNormalizedString());
+
+                var centralDependencyX = project1Spec.TargetFrameworks.First().CentralPackageVersions["x"];
+                var centralDependencyY = project1Spec.TargetFrameworks.First().CentralPackageVersions["y"];
+                var centralDependencyZ = project1Spec.TargetFrameworks.First().CentralPackageVersions["Z"];
+
+                Assert.Equal("x", centralDependencyX.Name);
+                Assert.Equal("[1.0.0-beta.*, )", centralDependencyX.VersionRange.ToNormalizedString());
```

Comment:
This is a comment on a design, but we should be explicit about how floating versions behave when a package reference is transitive. 

Specifically, floating versions are only allowed in direct references.

Reply:
I will propose the central floating when transitive. For this PR this is out of scope.